web/pdf_viewer.js

```diff
@@ -487,7 +489,16 @@ class PDFViewer {
     if (!this.pdfDocument) {
       return;
     }
-    this._setScale(val, false);
+    this._setScale(val, { noScroll: false });
+  }
+
+  setCurrentScaleValueWithOptions(val, options) {
```

Comment:
Do you expect any (future) direct *outside* calls of this method, since otherwise we should probably just make it private?